fix src/dst Properties in deep_copy(DynamicView,View)

tkordenbrock opened the pull request:
In deep_copy(DynamicView, View), fix the dst Properties (DP) and the src Properties (SP) that were swapped in the using statements defining dst_type and src_type.

Add a reproducer unit test.

dalg24-jenkins commented:
Can one of the admins verify this patch?

tkordenbrock commented:
@rppawlo

PhilMiller reviewed (approved):
Ok to test

crtrott commented:
Note this thing is segfaulting in testing.

tkordenbrock commented:
Hmm.  This passed my local testing with SERIAL, OPENMP and CUDA backends.  I'll look into it today.

crtrott commented:
Oh also you are targeting the wrong branch. We need this to be based on develop. Or you could retarget this at relase-candidate-3.7.02 and open another one with the corresponding change for develop.

tkordenbrock commented:
I fixed the segfault.  It was only showing up with the native code optimization turned on.

I also retargetted the PR to develop and rebased the source branch onto develop.

crtrott commented:
Retest this please.

dalg24 commented:
OK to test
